backend/datasource/instance_provider.go

```diff
@@ -59,12 +59,17 @@ func (ip *instanceProvider) GetKey(ctx context.Context, pluginContext backend.Pl
 		return nil, errors.New("data source instance settings cannot be nil")
 	}
 
-	defaultKey := pluginContext.DataSourceInstanceSettings.ID
-	if tID := tenant.IDFromContext(ctx); tID != "" {
-		return fmt.Sprintf("%s#%v", tID, defaultKey), nil
+	dsID := pluginContext.DataSourceInstanceSettings.ID
+
+	proxyHash := ""
+	proxyOpts, _ := pluginContext.DataSourceInstanceSettings.ProxyOptionsFromContext(ctx)
+	if proxyOpts != nil {
+		proxyHash = proxyOpts.ClientCfg.Hash()
 	}
```

Comment:
I think it would be nicer to add a dedicated function in the `backend` package for this. Something like:

```go
func ProxyHashFromContext(ctx context.Context) (string, bool) {
	cfg := GrafanaConfigFromContext(ctx)
	p, err := cfg.proxy()
	if err != nil {
		return "", false
	}
	return p.clientCfg.Hash(), true
}
```